### ISSUE TITLE
unarchive can't download archives when pipelining is enabled

### ISSUE THREAD

**Issue by ozamosi** [ansible/ansible-modules-core#2250]
##### Issue Type:

Bug Report
##### Ansible Version:

```
$ ansible --version
2.0.0
```

Latest git as of a few minutes ago, meaning ansible-modules-core 92d2501.
##### Ansible Configuration:

```
[defaults]
inventory = inventory

[ssh_connection]
pipelining = True
```
##### Environment:

Running on Ubuntu 15.04, against Debian Jessie.
##### Summary:

The unarchive module fails to unpack tar files retrieved from http when pipelining is enabled. It works fine when pipelining is disabled.
##### Steps To Reproduce:

```

---
- name: Test
  hosts: whatever
  remote_user: admin
  tasks:
    - unarchive:
        src: https://github.com/ansible/ansible/archive/v1.9.4-1.tar.gz
        dest: /tmp
        copy: no
```
##### Expected Results:

A new release of ansible in /tmp
##### Actual Results:

The unpack step fails with `{"cmd": "/bin/tar -xzf \"v1.9.4-1.tar.gz\"", "err": "tar (child): v1.9.4-1.tar.gz: Cannot open: No such file or directory\ntar (child): Error is not recoverable: exiting now\n/bin/tar: Child returned status 2\n/bin/tar: Error is not recoverable: exiting now\n", "out": "", "rc": 2}`

I did some digging, and on [https://github.com/ansible/ansible-modules-core/blob/92d2501/files/unarchive.py#L281](line 281 in unarchive.py), `__file__` is `<stdin>`, so `os.path.dirname(__file__)` returns `''`, which means the file is uploaded to `/home/admin` in my reproduction case, while `src` is set to the relative path `v1.9.4-1.tar.gz`, and `TgzArchive`'s `unarchive` specifies the `dest` directory as working dir and thus can't find the file.

If I disable pipelining, everything works fine, however.

**Comment by Jmainguy** [comment]
Thank you for reporting this, I was able to reproduce, once #2250 is merged this can be closed.

**Comment by abadger** [comment]
closing as this was fixed in #3307.
